robotoff/insights/importer.py

```diff
@@ -421,21 +444,53 @@ def get_insight_update(
                 cls.is_conflicting_insight(candidate, reference_insight)
                 for reference_insight in reference_insights
                 if reference_insight.annotation is not None
+                # Don't overwrite an insight that is going to be applied
+                # automatically soon
+                or reference_insight.automatic_processing is True
             )
 
             if not match:
-                for selected in to_create:
+                for selected, _ in to_create_or_update:
                     if cls.is_conflicting_insight(candidate, selected):
                         # Don't import candidate if it conflicts with an
                         # already selected candidate
                         break
                 else:
-                    to_create.append(candidate)
+                    mapping_ref_insight = None
+                    # In order for the voting system to work, we map insights to create
+                    # to existing insights with the same value/value_tag/source_image.
+                    # This way, we don't loose vote information.
+                    for reference_insight in reference_insights:
+                        if (
+                            reference_insight.annotation is None
+                            and cls.is_conflicting_insight(candidate, reference_insight)
+                            # only map to existing insight is the source image is the same,
+                            # otherwise create a new insight
```

Comment:
```suggestion
                            # only map to existing insight if the source image is the same,
                            # otherwise create a new insight
```